src/test/scala/GuessWhoGameSpec/BoardSpec.scala

```diff
@@ -0,0 +1,149 @@
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+    //Future testcases
+    //      gameBoard.getRemainingCharacters.forall(_.gender == "male") shouldBe false
+  }
+
+  it should "eliminate characters correctly based on name questions" in {
+    val gameBoard = new Board(characters)
+    val initialSize = gameBoard.remainingCharacters.size
+    gameBoard.handleQuestion("name", Left("muhammad"))
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+  }
+
+  it should "eliminate characters correctly based on hairColor questions" in {
+    val gameBoard = new Board(characters)
+    val initialSize = gameBoard.remainingCharacters.size
+    gameBoard.handleQuestion("hairColor", Left("blonde"))
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+  }
+
+  it should "eliminate characters correctly based on eyeColor questions" in {
+    val gameBoard = new Board(characters)
+    val initialSize = gameBoard.remainingCharacters.size
+    gameBoard.handleQuestion("eyeColor", Left("blue"))
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+  }
+
+  it should "eliminate characters correctly based on wearsGlasses questions" in {
+    val gameBoard = new Board(characters)
+    val initialSize = gameBoard.remainingCharacters.size
+    gameBoard.handleQuestion("wearsGlasses", Right(false))
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+  }
+
+  it should "eliminate characters correctly based on HasFacialHair questions" in {
+    val gameBoard = new Board(characters)
+    val initialSize = gameBoard.remainingCharacters.size
+    gameBoard.handleQuestion("hasFacialHair", Right(false))
+    val newSize = gameBoard.remainingCharacters.size
+    newSize should be < initialSize
+  }
+
+  it should "handle incorrect attribute values cleanly" in {
+    val gameBoard = Board(characters)
+    val initialSize = gameBoard.getRemainingCharacters.size
+    gameBoard.handleQuestion("gender", Left("nonexistent gender"))
+    val newSize = gameBoard.getRemainingCharacters.size
+
+    newSize shouldEqual initialSize
+  }
+
+  it should "handle edge case where all characters have the same attribute" in {
+    val sameGenderCharacters = List(
+      Character("john", "male", "brown", "blue", wearsGlasses = false, facialHair = false),
+      Character("paul", "male", "black", "green", wearsGlasses = true, facialHair = true)
+    )
+    val gameBoard = Board(sameGenderCharacters)
+    gameBoard.handleQuestion("gender", Left("male"))
+    gameBoard.getRemainingCharacters.size shouldEqual 2
+  }
+
+  it should "fail to eliminate characters with incorrect gender" in {
+    val gameBoard = Board(characters)
+    gameBoard.handleQuestion("gender", Left("female"))
+    gameBoard.getRemainingCharacters.forall(_.gender == "male") shouldBe true
+  }
```

Comment:
This test if failing. I can see what you are trying to do here, making sure your tests can 'fail' but we don't want the physical test failing, we want to make sure we can reach the unhappy cases (the error cases/the wildcards etc.)